### ISSUE TITLE
[rmodels] Attempting to load a simple OBJ model with 1 texture causes segmentation fault

### ISSUE THREAD

**Issue by keithstellyes** [raysan5/raylib#3398]
Please, before submitting a new issue verify and check:

 - [x] I tested it on latest raylib version from master branch
 - [x] I checked there is no similar issue already reported
 - [x] I checked the documentation on the [wiki](https://github.com/raysan5/raylib/wiki)
 - [x] My code has no errors or misuse of raylib

### Issue description

I have a simple model with a simple material that uses just an image. A segmentation fault is caused. When attempting load with `LoadModel()` I created the model in Blender along with the material. Not a `LoadMaterials` loads the material without crashing.

**EDIT:** JeffM suggested the use of the `gltf` format for models. This, thankfully, unblocks me and allows me to continue what I'm wanting to do. Regardless, still a bug here :)

### Environment

*Provide your Platform, Operating System, OpenGL version, GPU details where you experienced the issue.*

Arch Linux (`6.5.6-arch2-1`) NVIDIA GeForce RTX 3070 Ti

EGL API version: 1.5
EGL vendor string: NVIDIA
EGL version string: 1.5
EGL client APIs: OpenGL_ES OpenGL
OpenGL core profile vendor: NVIDIA Corporation
OpenGL core profile renderer: NVIDIA GeForce RTX 3070 Ti/PCIe/SSE2
OpenGL core profile version: 4.6.0 NVIDIA 535.113.01
OpenGL core profile shading language version: 4.60 NVIDIA
OpenGL compatibility profile vendor: NVIDIA Corporation
OpenGL compatibility profile renderer: NVIDIA GeForce RTX 3070 Ti/PCIe/SSE2
OpenGL compatibility profile version: 4.6.0 NVIDIA 535.113.01
OpenGL compatibility profile shading language version: 4.60 NVIDIA

### Issue Screenshot

N/A

See attachment for the model (OBJ), the material, and an image that the material depends on. May need to modify the `.mtl` to get raylib to find it. Additionally, I have attached the Blender file used to create the model in case it helps investigation (but is not needed to reproduce the issue)
[files.zip](https://github.com/raysan5/raylib/files/12863782/files.zip)

Note that the model looks correct w/o issue on an online viewer https://3dviewer.net/

![image](https://github.com/raysan5/raylib/assets/13317074/7fe12972-039e-4259-827d-8de09fc2883d)

`free -g` output:

```
               total        used        free      shared  buff/cache   available
Mem:              31           4          24           0           2          26
Swap:              0           0           0
```
### Code Example

```c
InitWindow(800, 450, "Fly!");
Model levelMesh = LoadModel("levelmesh.obj");
```

Log w/ valgrind:

```
==6057== Invalid read of size 4
==6057==    at 0x18F1C1: LoadOBJ (rmodels.c:3981)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0x9d8555c is 4 bytes before a block of size 4 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F16D: LoadOBJ (rmodels.c:3972)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F1C6: LoadOBJ (rmodels.c:3981)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0x9d8555c is 4 bytes before a block of size 4 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F16D: LoadOBJ (rmodels.c:3972)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F5BF: LoadOBJ (rmodels.c:4023)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6a60 is 0 bytes after a block of size 432 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F351: LoadOBJ (rmodels.c:4005)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F69D: LoadOBJ (rmodels.c:4024)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6a6c is 12 bytes after a block of size 432 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F351: LoadOBJ (rmodels.c:4005)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F77B: LoadOBJ (rmodels.c:4025)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6a78 is 24 bytes after a block of size 432 in arena "client"
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F846: LoadOBJ (rmodels.c:4032)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6bc0 is 0 bytes after a block of size 288 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F3AC: LoadOBJ (rmodels.c:4006)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F8C7: LoadOBJ (rmodels.c:4033)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6bc4 is 4 bytes after a block of size 288 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F3AC: LoadOBJ (rmodels.c:4006)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F970: LoadOBJ (rmodels.c:4034)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6bc8 is 8 bytes after a block of size 288 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F3AC: LoadOBJ (rmodels.c:4006)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18F9F1: LoadOBJ (rmodels.c:4035)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6bcc is 12 bytes after a block of size 288 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F3AC: LoadOBJ (rmodels.c:4006)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18FA9A: LoadOBJ (rmodels.c:4036)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6bd0 is 16 bytes after a block of size 288 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F3AC: LoadOBJ (rmodels.c:4006)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18FB1B: LoadOBJ (rmodels.c:4037)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6bd4 is 20 bytes after a block of size 288 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F3AC: LoadOBJ (rmodels.c:4006)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18FBF3: LoadOBJ (rmodels.c:4043)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6db0 is 0 bytes after a block of size 432 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F40B: LoadOBJ (rmodels.c:4007)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 
==6057== Invalid write of size 4
==6057==    at 0x18FCD1: LoadOBJ (rmodels.c:4044)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057==  Address 0xa0b6dbc is 12 bytes after a block of size 432 alloc'd
==6057==    at 0x48469B3: calloc (vg_replace_malloc.c:1554)
==6057==    by 0x18F40B: LoadOBJ (rmodels.c:4007)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
==6057== 

valgrind: m_mallocfree.c:304 (get_bszB_as_is): Assertion 'bszB_lo == bszB_hi' failed.
valgrind: Heap block lo/hi size mismatch: lo = 496, hi = 4777118717294609368.
This is probably caused by your program erroneously writing past the
end of a heap block and corrupting heap metadata.  If you fix any
invalid writes reported by Memcheck, this assertion failure will
probably go away.  Please try that before reporting this as a bug.


host stacktrace:
==6057==    at 0x58042C9A: show_sched_status_wrk (m_libcassert.c:406)
==6057==    by 0x58042DC7: report_and_quit (m_libcassert.c:477)
==6057==    by 0x58042F6B: vgPlain_assert_fail (m_libcassert.c:543)
==6057==    by 0x5804CCD8: get_bszB_as_is (m_mallocfree.c:304)
==6057==    by 0x5804CCD8: is_inuse_block (m_mallocfree.c:332)
==6057==    by 0x5804CCD8: vgPlain_describe_arena_addr (m_mallocfree.c:1614)
==6057==    by 0x5803B24A: vgPlain_describe_addr (m_addrinfo.c:185)
==6057==    by 0x5803996E: vgMemCheck_update_Error_extra (mc_errors.c:1224)
==6057==    by 0x5803E2B5: vgPlain_maybe_record_error (m_errormgr.c:822)
==6057==    by 0x58038B58: vgMemCheck_record_address_error (mc_errors.c:795)
==6057==    by 0x100807E04B: ???
==6057==    by 0x1002CB5F1F: ???

sched status:
  running_tid=1

Thread 1: status = VgTs_Runnable (lwpid 6057)
==6057==    at 0x18FDAF: LoadOBJ (rmodels.c:4045)
==6057==    by 0x17C8D0: LoadModel (rmodels.c:1037)
==6057==    by 0x10CD2F: main (main.cpp:51)
client stack range: [0x1FFEFF9000 0x1FFF000FFF] client SP: 0x1FFEFFF7B0
valgrind stack range: [0x1002BB6000 0x1002CB5FFF] top usage: 15216 of 1048576
```

**Comment by raysan5** [comment]
@keithstellyes Are you exporting the OBJ with triangulated faces, afair, raylib OBJ importer does not support quads.

**Comment by keithstellyes** [comment]
It looks like there's indeed quads, I'll have to try again without the quads

**Comment by raysan5** [comment]
@keithstellyes I think so but I don't know if it is actually working...

**Comment by raysan5** [comment]
Reviewed this issue and it seems code was modified in commit (https://github.com/raysan5/raylib/commit/e8fa7ceb79613188c495b80a332875a1afdab8cf) to support materials per faces, it added extra logic complexity and now it breaks for some specific situations like this one.

I think the best solution is rewrite and simplify that code but it requires some work...

**Comment by keithstellyes** [comment]
Sorry have been busy, I'll attempt to rerun it again this week ☺️ I will
look into this, (maybe even see if I can try to fix it, since I've been
lately trying to learn more about the lower level stuff)

**Comment by raysan5** [comment]
@keithstellyes No worries, I just reviewed and fixed it. Actually `LoadOBJ()` was redesigned and now triangulated meshes based on quads are properly supported. Note that on the provided sample file (.obj) the texture coordinates are not correctly defined.
